app/validators/rent_validator.rb

```diff
@@ -0,0 +1,102 @@
+class RentValidator < ActiveModel::Validator
```

Comment:
This class is long but it also does complex work. I think it's quite easy to read. Good work.